Lib/test/test_netrc.py

```diff
@@ -123,6 +124,33 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
+            self.assertRaises(FileNotFoundError, netrc.netrc)
+
+    def test_file_not_found_explicit(self):
+        self.assertRaises(FileNotFoundError, netrc.netrc,
+                          file="unlikely_netrc")
+
+    def test_home_not_set(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        fake_netrc_path = os.path.join(d, 'netrc')
+        with open(fake_netrc_path, 'w') as f:
+            f.write("machine foo.domain.com login bar password pass")
+        os.chmod(fake_netrc_path, 0o600)
+        with support.EnvironmentVarGuard() as environ:
+            environ.unset('HOME')
+            with mock.patch('os.path.expanduser') as mock_expanduser:
```

Comment:
I'm afraid not. If we unset HOME and `~/.netrc` *does* exist on the developer box, `os.path.expanduser` will read `/etc/passwd` using `pwd` `(see [1]`) and no exception will be raised. I'd prefer the tests to not depend on what's on the home directory of the user running the tests.

`[1]`: https://github.com/python/cpython/blob/master/Lib/posixpath.py#L245-L247